File: shared/src/commonMain/kotlin/com/jetbrains/swiftinteropplayground/classes/OpenClass.kt
package com.jetbrains.swiftinteropplayground.classes

open class OpenClassWithConstructorParams(
    val param1: String,
    val param2: Boolean
) {
    protected val someField: String get() = "14"

    fun finalFunctionInClass() {
        println("Final method in open class")
    }

    open fun functionCanBeOverridden() {
        println("Open method in open class")
    }

}

File: iosApp/iosApp/Classes/OpenClass.swift
import Foundation
import shared

class OpenClassExample : OpenClassWithConstructorParams {
    override func finalFunctionInClass() {
        print("Final function was overridden")
    }
    
    override func functionCanBeOverridden() {
        print("Open function was overridden")
    }
    
    static func example() {
        let example = OpenClassExample(param1: "123", param2: true)
        let _ = example.someField
        example.finalFunctionInClass()
        example.functionCanBeOverridden()
    }
}
